biz.aQute.launcher/src/aQute/launcher/pre/EmbeddedLauncher.java

```diff
@@ -49,18 +47,19 @@ public static void main(String... args) throws Exception {
 			String runpath = m.getMainAttributes()
 				.getValue(EMBEDDED_RUNPATH);
 			if (runpath != null) {
-				MANIFEST = m;
-				List<URL> classpath = new ArrayList<>();
+				try (ModifiableURLClassLoader ml = new ModifiableURLClassLoader(cl)) {
+					for (String path : runpath.split("\\s*,\\s*")) {
+						URL url = toFileURL(cl.getResource(path));
+						ml.addURL(url);
+					}
 
-				for (String path : runpath.split("\\s*,\\s*")) {
-					URL url = toFileURL(cl.getResource(path));
-					classpath.add(url);
-				}
+					Class<?> embeddedLauncher = ml.loadClass("aQute.launcher.pre.EmbeddedLauncher");
```

Comment:
You are loading this class again from the new class loader and setting MANIFEST? That seems only necessary because of my earlier concern about `ModifiableURLClassLoader` not delegating to parent first, so `ModifiableURLClassLoader` loads a new EmbeddedLauncher when Launcher want to access it's MANIFEST field. So why not parent first?

Reply:
This is the only solution. The Launcher class MUST NOT be on the parent classloader because it's not modifiable.